x-pack/osquerybeat/ext/osquery-extension/pkg/encoding/encoding.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package encoding
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+type EncodingFlag int
+
+const (
+	EncodingFlagParseUnexported EncodingFlag = 1 << iota
+	EncodingFlagUseNumbers0Values
```

Comment:
Let's spell out [zero value](https://go.dev/ref/spec#The_zero_value). I think that is more conventional.

`s/EncodingFlagUseNumbers0Values/EncodingFlagUseNumbersZeroValues/g`